Error while checking build/O2/fullCI for a77366d6c5e20e2cba033070e32edcf257dd3680 at 2024-07-17 06:07:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
/sw/SOURCES/O2/13302-slc8_x86-64/0/Detectors/ITSMFT/common/reconstruction/include/ITSMFTReconstruction/AlpideCoder.h:674:41: error: no member named 'getAPEByte' in 'o2::itsmft::ChipStat'
Error: /sw/slc8_x86-64/ROOT/v6-30-01-alice5-20/bin/rootcling: compilation failure (/sw/BUILD/2ffb439c7791ddeb2443337f838a0ad27398f093/O2/Detectors/ITSMFT/common/reconstruction/G__O2ITSMFTReconstructionbcc6992db9_dictUmbrella.h)
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/13302/a77366d6c5e20e2cba033070e32edcf257dd3680/build_O2_fullCI/pretty.html).

--- a/Detectors/ITSMFT/common/reconstruction/include/ITSMFTReconstruction/AlpideCoder.h
+++ b/Detectors/ITSMFT/common/reconstruction/include/ITSMFTReconstruction/AlpideCoder.h
@@ -659,6 +659,7 @@ class AlpideCoder
         case ChipStat::UnknownWord:
         case ChipStat::RepeatingPixel:
         case ChipStat::WrongRow:
+          break;
         case ChipStat::APE_STRIP_START:
         case ChipStat::APE_ILLEGAL_CHIPID:
         case ChipStat::APE_DET_TIMEOUT:
@@ -669,7 +670,24 @@ class AlpideCoder
         case ChipStat::APE_PENDING_DETECTOR_EVENT_LIMIT:
         case ChipStat::APE_PENDING_LANE_EVENT_LIMIT:
         case ChipStat::APE_O2N_ERROR:
-        case ChipStat::APE_RATE_MISSING_TRG_ERROR:
+        case ChipStat::APE_RATE_MISSING_TRG_ERROR: {
+          uint8_t errorByte = ChipStat::getAPEByte((ChipStat::DecErrors)errIdx);
+          if (dataRaw == errorByte) {
+            buffer.next(dataRaw); // Skipping error byte
+            // If we encountered the byte corresponding to the APE error,
+            // check that the rest of the raw stream consists of only
+            // padding.
+            while (buffer.next(dataRaw)) {
+              if (dataRaw != 0x00) {
+                break;
+              }
+            }
+            if (buffer.isEmpty()) {
+              res = VerifierMismatchResult::EXPECTED_MISMATCH;
+            }
+          }
+          break;
+        }
         case ChipStat::APE_PE_DATA_MISSING:
         case ChipStat::APE_OOT_DATA_MISSING:
         case ChipStat::WrongDColOrder:

--- a/Detectors/ITSMFT/common/reconstruction/include/ITSMFTReconstruction/DecodingStat.h
+++ b/Detectors/ITSMFT/common/reconstruction/include/ITSMFTReconstruction/DecodingStat.h
@@ -174,6 +174,15 @@ struct ChipStat {
     ft = c >= 0xf2 && c <= 0xfe;
     return APE_STRIP_START + c - 0xf2;
   }
+
+  // return APE byte that corresponds to the given APE DecErrors
+  static uint8_t getAPEByte(DecErrors c)
+  {
+    if (c < APE_STRIP_START || c > APE_OOT_DATA_MISSING) {
+      return 0xFF;
+    }
+    return 0xF2 + c - APE_STRIP_START;
+  }
   uint32_t getNErrors() const;
   uint32_t addErrors(const ChipPixelData& d, int verbosity);
   void print(bool skipNoErr = true, const std::string& pref = "FEEID") const;

